src/Yasumi/Yasumi.php

```diff
@@ -186,6 +186,44 @@ public static function create($class, $year = null, $locale = self::DEFAULT_LOCA
     }
 
     /**
+     * Create a new holiday provider instance.
+     *
+     * A new holiday provider instance can be created using this function. You can use one of the providers included
+     * already with Yasumi, or your own provider by giving the name of your class in the first parameter. Your provider
+     * class needs to implement the 'ProviderInterface' class.
+     *
+     * @param string $iso3166_2  ISO3166-2 Coded region, holiday provider will be searched for
+     * @param int    $year   year for which the country provider needs to be created. Year needs to be a valid integer
+     *                       between 1000 and 9999.
+     * @param string $locale The locale to use. If empty we'll use the default locale (en_US)
+     *
+     * @throws RuntimeException         If no such holiday provider is found
+     * @throws InvalidArgumentException if the year parameter is not between 1000 and 9999
+     * @throws UnknownLocaleException   if the locale parameter is invalid
+     * @throws InvalidArgumentException if the holiday provider for the given country does not exist
+     *
+     * @return AbstractProvider An instance of class $class is created and returned
+     */
+    public static function createByISO3166_2($iso3166_2, $year = null, $locale = self::DEFAULT_LOCALE)
+    {
+        $providers = self::getProviders();
+
+        $class = isset($providers[$iso3166_2])
```

Comment:
You can simplify this by combining this and the '`if ($class === false)`' into one. Then no need for the intermediate variable `$class` and ternary check.